src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseReader.cs

```diff
@@ -832,6 +832,12 @@ public override XmlNodeType MoveToContent()
                 {
                     if (_node.NodeType != XmlNodeType.Text && _node.NodeType != XmlNodeType.CDATA)
                         break;
+
+                    if (_trailByteCount > 0)
+                    {
+                        break;
+                    }
+
```

Comment:
I had the same concern. I checked the history of the desktop.  It seems we have two code bases for serialization. The CoreFx code was copied from NDP\FX\src\CDF, which doesn't have this fix. The other one is under NDP\CDF\src, and it has the fix. Do you know the relationships of these two code base? @iamjasonp